### ISSUE TITLE
Error: invalid json response body

### ISSUE THREAD

**Issue by yaroslavsadin** [microsoft/vscode-docker#3185]
Hi,

I can't connect to our company's private registry on Nexus. I still can connect using 1.10.0 version but not with the later. I checked that `/v2/` returns `200` but Docker says:

```
Error: invalid json response body at https://.../v2/ reason: Unexpected end of JSON input
```

Could you give me a clue on what had changed?

**Comment by bwateratmsft** [comment]
It looks like in the changeover from `request` to `node-fetch`, the behavior for empty bodies is different. The `/v2/` body is empty in some cases and should parse into an `undefined` JSON object.

**Comment by bwateratmsft** [comment]
The fix for this has been released in Docker extension 1.17.0.
